Working with your own data document (#195)
AlexsLemonade/training-modules: virtual-setup/working-with-your-own-data.md

The most simple `wget` command just needs the URL to pull the file from.

*Template:*
```
wget <URL>
```

*Specific example:* Here's an example of us downloading a file from ArrayExpress
```
wget https://www.ebi.ac.uk/arrayexpress/files/E-GEOD-67851/E-GEOD-67851.processed.1.zip
```

By default, the file will be saved to the current directory and the file name it had from its origin (so with the above example `E-GEOD-67851.processed.1.zip`).

Likely you will want to be more specific about where you are saving the file to and what you are calling it.
For that, we can use the `-O`, or `output` option with our `wget` command and specify a file path.  

*Template:*
```
wget -O <FILE_PATH_TO_SAVE_TO> <URL>
```

*Specific example:* Here's an example where we will download that same array express file, but instead save it to the `data` folder and call it `some_array_data.zip`.
(Best to keep the file extension consistent to avoid troubles!)

```
wget -O data/some_array_data.zip https://www.ebi.ac.uk/arrayexpress/files/E-GEOD-67851/E-GEOD-67851.processed.1.zip
```

`-O` is one of many `wget` command options.

> To see the complete list of `wget` options, use the command: `wget -h` in bash.

Reviewer: ```suggestion
To see the complete list of `wget` options, use the command: `wget -h` in Terminal.
```
This is to keep it consistent with the language you used above.

![Upload button](screenshots/upload-button.png)

A mini screen will pop up asking you to choose the file you want to upload:

![Choose file](screenshots/upload-choose-file.png)

Choose your compressed data file, and click `OK`.
This may take some time, particularly if you have a large dataset.  

When the server is finished uploading your data, you should see your file in your `home` directory!
It will automatically be uncompressed.

### Reading the data into your R environment.

This step is very dependent on the format of your data and what you are planning to do with it!
If your file is a TSV, CSV, or RDS file, follow the examples in the [`intro-to-R-tidyverse/intro-to-tidyverse.Rmd` notebook](intro-to-R-tidyverse/03-intro_to_tidyverse.Rmd).

We do NOT recommend clicking on the file in the RStudio panel to load it into your R environment, this typically won't work for anything that's not a very small file.

Plus, for reproducibility purposes, you should write the data reading step into your notebook analyses!

## Downloading files

Any files on the RStudio server you would like to save to your computer you can export.

> **Step 1)** Use the Export button!

Reviewer 2: ```suggestion
**Step 1)** Select the file(s) or folder(s) you would like to download
Check the box(es) to the left of the files or folder(s) in the **Files** pane.

**Step 2)** Use the Export button!
```

Click on the `More` button with a gear next to it in the lower right pane.

![Export button](screenshots/export-button.png)

> **Step 2)** Specify the file you'd like to download.

Reviewer 2: ```suggestion
**Step 3)** Specify the name you would like the downloaded file to have.
```

![Export window](![Export button](screenshots/export-window.png))

> **Step 3)** Find where the file downloaded.

Reviewer 2: ```suggestion
**Step 4)** Find where the file downloaded.
```

```

By default, the file will be saved to the current directory and the file name it had from its origin (so with the above example `E-GEOD-67851.processed.1.zip`).

Likely you will want to be more specific about where you are saving the file to and what you are calling it.
For that, we can use the `-O`, or `output` option with our `wget` command and specify a file path.  

*Template:*
```
wget -O <FILE_PATH_TO_SAVE_TO> <URL>
```

*Specific example:* Here's an example where we will download that same array express file, but instead save it to the `data` folder and call it `some_array_data.zip`.
(Best to keep the file extension consistent to avoid troubles!)

```
wget -O data/some_array_data.zip https://www.ebi.ac.uk/arrayexpress/files/E-GEOD-67851/E-GEOD-67851.processed.1.zip
```

`-O` is one of many `wget` command options.
To see the complete list of `wget` options, use the command: `wget -h` in bash.

[See more `wget` examples](https://www.tecmint.com/10-wget-command-examples-in-linux/)

### Upload data from your own computer

If the data you want to use is stored locally on your computer, here's how we recommend uploading it to the RStudio Server.

> **Step 1)** We recommend you compress your data folder into a single zip file.

Reviewer 2: Note limit for browser uploads to the server is probably somewhere around 1GB, so they may need to split things up, or ask for help.

For most operating systems, you can right-click on your data folder, and choose `Compress` to zip up your files
- [Windows zipping](https://support.microsoft.com/en-us/help/14200/windows-compress-uncompress-zip-files)
- [Mac zipping](https://www.imore.com/how-compress-file-your-mac)

For reference, here's how you [compress files from the command line](https://coolestguidesontheplanet.com/how-to-compress-and-uncompress-files-and-folders-in-os-x-lion-10-7-using-terminal/).

**Step 2)** Once your data is compressed to a single file, [navigate to your RStudio session](./rstudio-login.md).

**Step 3)** Use the `Upload button` to choose your compressed data folder.

This button is in the lower right panel of your RStudio session:
![Upload button](screenshots/upload-button.png)

A mini screen will pop up asking you to choose the file you want to upload:

![Choose file](screenshots/upload-choose-file.png)

Choose your compressed data file, and click `OK`.
This may take some time, particularly if you have a large dataset.  

> When the server is finished uploading your data, you should see your file in your `home` directory!

Reviewer 2: Is it automatically uncompressed? If not, do we direct to do so?

Author: Yes. It's automatically uncompressed whether we like it or not.

Author: I can add that info on here so people are aware.

Reviewer 2: Github does have an emoticon that captures my feelings about this information.

### Reading the data into your R environment.

This step is very dependent on the format of your data and what you are planning to do with it!
If your file is a TSV, CSV, or RDS file, follow the examples in the [`intro-to-R-tidyverse/intro-to-tidyverse.Rmd` notebook](intro-to-R-tidyverse/03-intro_to_tidyverse.Rmd).

We do NOT recommend clicking on the file in the RStudio panel to load it into your R environment, this typically won't work for anything that's not a very small file.

Plus, for reproducibility purposes, you should write the data reading step into your notebook analyses!

## Downloading files

Any files on the RStudio server you would like to save to your computer you can export.

> **Step 1)** We recommend you compress the files you want copied to your computer into a zip file.

Reviewer 2: The server will automatically do this step (at least for folders or multiple files). They do not need to zip on their own.

Author: Oh! Good to know!

[See more `wget` examples](https://www.tecmint.com/10-wget-command-examples-in-linux/)

As is recommended and also shown with this example, this dataset is zipped.
This means after you successfully `wget` the file, you will need to unzip it.
To unzip the contents to a particular directory, we will use the `-d` option.

*Template*
```
unzip -d <DIRECTORY_TO_UNZIP_TO> <FILE_TO_UNZIP>
```

*Specific example:*
Here we will unzip the contents of data/some_array_data.zip to be saved to the directory `data/`.

> First, however, we'll need to make the directory `data` using a command `mkdir`.

Reviewer 2: if they downloaded it already to `data/`, they don't need to recreate it. `unzip` will also create directories, at least if they are only one level deep from currently existing directories.
```suggestion
```

```

> mkdir data
> 

Reviewer 2: ```suggestion
```
